### ISSUE TITLE
Pass execution context to table functions for accessing external datasource

### ISSUE THREAD

**Issue by zhicwu** [ClickHouse/ClickHouse#33803]
Does it make sense to pass execution context to table functions for accessing external datasource?

Taking `url` and `jdbc` as example, since they're based on http protocol, it would be great if ClickHouse can put below headers in **request**. With that, extension like JDBC bridge can implement role-based access control and detect recursive query like mentioned in #33802. 

| Header | Mandatory | Description |
| ------- | ----------- | ----------- |
| User-Agent | N | server/lib + version |
| X-ClickHouse-Server-Display-Name | N | server display name |
| X-ClickHouse-Format | N | different default value for different table function |
| X-ClickHouse-Query-Id | Y | query id |
| X-ClickHouse-User | Y | user who issued the query on ClickHouse, it could be user name, id or even a hash for security reason |
| X-ClickHouse-Timezone | Y | server time zone |
| X-ClickHouse-Version | N |server version |
| X-ClickHouse-Revision | N | server revision |

Note: more headers can be added for tracing when needed.

**Comment by alexey-milovidov** [comment]
Yes, it makes sense. We already do with User-Agent IIRC. We should also pass OpenTelemetry headers.
